src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -389,34 +392,34 @@ internal static bool IsMinimalProgressRenderingEnabled()
 
             // if the activity is really long, only use up to half the width
             string activity;
-            if (Activity.Length > maxWidth / 2)
+            if (rawUI.LengthInBufferCells(Activity) > maxWidth / 2)
             {
-                activity = Activity.Substring(0, maxWidth / 2) + PSObjectHelper.Ellipsis;
+                activity = StringUtil.TruncateToBufferCellWidth(rawUI, Activity, maxWidth / 2) + PSObjectHelper.Ellipsis;
             }
             else
             {
                 activity = Activity;
             }
 
             // 4 is for the extra space and square brackets below and one extra space
-            int barWidth = maxWidth - activity.Length - indentation - 4;
+            int barWidth = maxWidth - rawUI.LengthInBufferCells(activity) - indentation - 4;
 
             var sb = new StringBuilder();
             int padding = maxWidth + PSStyle.Instance.Progress.Style.Length + PSStyle.Instance.Reverse.Length + PSStyle.Instance.ReverseOff.Length;
             sb.Append(PSStyle.Instance.Reverse);
 
             int maxStatusLength = barWidth - secRemainLength - 1;
-            if (maxStatusLength > 0 && StatusDescription.Length > barWidth - secRemainLength)
+            if (maxStatusLength > 0 && rawUI.LengthInBufferCells(StatusDescription) > barWidth - secRemainLength)
             {
-                sb.Append(StatusDescription.AsSpan(0, barWidth - secRemainLength - 1));
+                sb.Append(StringUtil.TruncateToBufferCellWidth(rawUI, StatusDescription, maxStatusLength));
                 sb.Append(PSObjectHelper.Ellipsis);
             }
             else
             {
                 sb.Append(StatusDescription);
-            }
+            }   
 
-            int emptyPadLength = barWidth + PSStyle.Instance.Reverse.Length - sb.Length - secRemainLength;
+            int emptyPadLength = barWidth - rawUI.LengthInBufferCells(sb.ToString()) - secRemainLength;
```

Comment:
It looks that `sb` contains VT sequences, does `LengthInBufferCells` strip off VT sequences when counting cells?

Reply:
`LengthInBufferCells` seems to only handle double-wide chars and nothing with VT.  It seems that we shouldn't allow VT in the status description since it would screw up rendering anyways, so I'll add a change that will error if VT is in the status description.

Reply:
It's good to prevent a user from using VT decorated `Status` and `Activity`.
However, here we already appended `PSStyle.Instance.Reverse` to the `sb`, so the returned result from `rawUI.LengthInBufferCells(sb.ToString())` will be incorrect. You will need the original `+ PSStyle.Instance.Reverse.Length`.

Or, maybe consider to not add the `Reverse` prefix until we are wrapping up constructing `sb` at the line 440 and 445, where `ReverseOff` VT sequence is added. That will make the code clearer and more readable.

Reply:
`PSStyle.Instance.Reverse` should have a buffer cell length of zero since it shouldn't have any printable characters.  So the length of it shouldn't be used at all.  Removed it in the calculation of the length of the bar.

Reply:
Exactly, it should have a buffer cell length of zero, but the returned value from `rawUI.LengthInBufferCells(sb.ToString())` will include the length of `PSStyle.Instance.Reverse`, isn't it
I think that's why `+ PSStyle.Instance.Reverse.Length` was there to offset it.